File: src/main/java/com/snackgame/server/auth/oauth/oidc/payload/GoogleIdTokenPayload.java
package com.snackgame.server.auth.oauth.oidc.payload;

import java.util.Map;

public class GoogleIdTokenPayload implements IdTokenPayload {

    private final Map<String, Object> payload;

    public GoogleIdTokenPayload(Map<String, Object> payload) {
        this.payload = payload;
    }

    @Override
    public String getProvider() {
        return "google";
    }

    @Override
    public String getId() {
        return payload.get("sub").toString();
    }

    @Override
    public String getEmail() {
        return payload.get("email").toString();
    }
}

File: src/main/java/com/snackgame/server/auth/oauth/oidc/payload/GoogleIdTokenPayload.kt
package com.snackgame.server.auth.oauth.oidc.payload

class GoogleIdTokenPayload(private val payload: Map<String, Any>) : IdTokenPayload {
    override val provider: String
        get() = "google"
    override val id: String
        get() = payload["sub"] as String
    override val email: String
        get() = payload["email"] as String
    override val name: String?
        get() = payload["name"] as String?
    override val picture: String?
        get() = payload["picture"] as String?
}
